systems-management/nisysmgmt.yml

```diff
@@ -0,0 +1,2030 @@
+        $ref: '#/definitions/ManagedSystemDataProperty'
+        example:
+          last_updated_timestamp: 2019-02-19T11:42:25.078Z
+          data:
+            - resource_uri: nisyscfg//sys/bus/pci/devices/0000:04:00.0
+              properties:
+              - type: string
+                tag: 16777216
+                value: "//localhost/nisyscfg/%2Fsys%2Fbus%2Fpci%2Fdevices%2F0000:04:00.0"
+              - type: bool
+                tag: 16781312
+                value: true
+            - resource_uri: nisyscfg/enp0s25
+              properties:
+              - type: string
+                tag: 16777216
+                value: "//localhost/nisyscfg/enp0s25"
+              - type: bool
+                tag: 16781312
+                value: false
+      groups:
+        description: Groups information about a system. Specifies on which groups is the system assigned to.
+        $ref: '#/definitions/ManagedSystemDataProperty'
+        example:
+          last_updated_timestamp: 2019-02-19T11:42:25.078Z
+          data:
+            - 121b471c-d7d4-4996-a38f-27f7e58be070
+      keywords:
+        description: Keywords metadata information about a system.
+        type: object
+        properties:
+          last_updated_timestamp:
+            description: Last updated timestamp.
+            type: string
+            format: iso-date-time
+            example: 2019-02-19T11:42:25.078Z
+          data:
+            description: A list of strings defining metadata information about a system.
+            type: array
+            items:
+              type: string
+              example: 
+                - testVM
+        required: [last_updated_timestamp, data]
+      properties:
+        description: Properties metadata information about a system.
+        type: object
+        properties:
+          last_updated_timestamp:
+            description: Last updated timestamp.
+            type: string
+            format: iso-date-time
+          data:
+            description: An object defining key-value pairs metadata information about a system.
+            type: object
+            additionalProperties:
+              type: string
+            example:
+              owner: admin
+        required: [last_updated_timestamp, data]
+    required: [id, created_timestamp, last_updated_timestamp, connected]
+
+  ReportType:
+    title: Report Type
+    type: string
+    enum:
+      - Software
+      - Hardware
+    example: Software
+    description: >
+      The type of the report to be created.
+       * Software: The report will contain the installed software packages.
+       * Hardware: The report will contain information about the hardware configuration of the system.
+
+  SystemsReportRequest:
+    description: Request containing the report type followed by the list of system IDs.
+    title: Systems Report Request
+    type: object
+    properties:
+      type:
+        $ref: '#/definitions/ReportType'
+      ids:
```

Comment:
Does this need to be an array of IDs, or could this use the new query syntax as a filter for any matching systems, including a list of IDs?

Reply:
good catch!